sale_order_restrict_cancel_existing_invoice/models/sale_order.py

```diff
@@ -10,24 +10,24 @@ class SaleOrder(models.Model):
     _inherit = "sale.order"
 
     confirmed_invoice_ids = fields.Many2many(
-        comodel_name="account.invoice",
+        comodel_name="account.move",
         compute="_compute_confirmed_invoice_ids",
         help="Technical field in order to retrieve confirmed invoices",
     )
 
     def _get_confirmed_invoice_states(self):
         """
-        Return the invoice states that should avoid order cancel
+        Return the invoice move_type that should avoid order cancel
         :return:
         """
-        return ["open"]
+        return ["out_invoice"]
```

Comment:
Should be 'posted'

Reply:
Ok, the function name is confusing.

Could you update it ?

Reply:
done